eclair-core/src/main/scala/fr/acinq/eclair/blockchain/electrum/db/sqlite/SqliteWalletDb.scala

```diff
@@ -197,13 +197,18 @@ object SqliteWalletDb {
 
   /**
     * change this value
-    * -if the new codec is incompatible with the old one
+    * - if the new codec is incompatible with the old one
     * - OR if you want to force a full sync from Electrum servers
     */
-  val version = 0x0000
+  val legacyVersion = 0x0000
+  val bech32Version = 0x0001
+
+  val walletTypeCodec = discriminated[WalletType].by(uint32)
+    .typecase(legacyVersion, provide(P2SH_SEGWIT))
+    .typecase(bech32Version, provide(NATIVE_SEGWIT))
 
   val persistentDataCodec: Codec[PersistentData] = (
-    ("version" | constant(BitVector.fromInt(version))) ::
+    ("walletType" | walletTypeCodec) ::
```

Comment:
`version` here is not supposed to encoded a specific property, but the version of the persistence format. Changing it means that legacy cannot be read, and in this case the onchain wallet behaves as if it had been restored from seed (it is equivalent to resetting the wallet cache from the wallet GUI). 
Here we need a breaking change (add a new `walletType` field) so I would just set `version` to 1 and add the new field to the codec